integration/helpers.go

```diff
@@ -1022,19 +1022,20 @@ type proxyServer struct {
 func (p *proxyServer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	// Validate http connect parameters.
 	if r.Method != http.MethodConnect {
-		trace.WriteError(w, trace.BadParameter("%v not supported", r.Method))
+		trace.WriteError(w, trace.Unwrap(trace.BadParameter("%v not supported", r.Method)))
 		return
 	}
 	if r.Host == "" {
-		trace.WriteError(w, trace.BadParameter("host not set"))
+		trace.WriteError(w, trace.Unwrap(trace.BadParameter("host not set")))
 		return
 	}
 
 	// Dial to the target host, this is done before hijacking the connection to
 	// ensure the target host is accessible.
 	dconn, err := net.Dial("tcp", r.Host)
 	if err != nil {
-		trace.WriteError(w, err)
+		log.WithError(err).Info(err.Error())
```

Comment:
This should be at least `Warn` and maybe be provide a reason for failure?
```go
log.WithFields(logrus.Fields{
	logrus.ErrorKey: err,
	"addr": r.Host,
}).Warn("Failed to dial.")
```
also a couple of Info->Warn below.